src/Nocks/Provider.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace SocialiteProviders\Nocks;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'NOCKS';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['user.read'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getWebsiteUrl().'oauth/authorize', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return $this->getWebsiteUrl().'oauth/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get($this->getApiUrl().'user', [
+            'headers' => [
+                'Authorization' => 'Bearer '.$token,
+                'Accept'        => 'application/json',
+            ],
+        ]);
+
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'                => $user['data'][0]['uuid'],
```

Comment:
I'd set `$data = $user['data'][0]` to avoid having to repeat it on every single line